saleor/graphql/discount/types/sales.py

```diff
@@ -127,80 +125,132 @@ class Meta:
             + " Use `Promotion` type instead."
         )
         interfaces = [relay.Node, ObjectWithMetadata]
-        model = models.Sale
+        model = models.Promotion
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+    @staticmethod
+    def resolve_created(root: ChannelContext[models.Promotion], _info: ResolveInfo):
+        return root.node.created_at
 
     @staticmethod
-    def resolve_created(root: models.Sale, _info: ResolveInfo):
-        return root.created_at
+    def resolve_type(root: ChannelContext[models.Promotion], info: ResolveInfo):
+        def _get_type(rules):
+            # We ensure, that old sales have at least one rule associated.
+            return rules[0].reward_value_type
+
+        return (
+            PromotionRulesByPromotionIdLoader(info.context)
+            .load(root.node.id)
+            .then(_get_type)
+        )
 
     @staticmethod
     def resolve_categories(
-        root: ChannelContext[models.Sale], info: ResolveInfo, **kwargs
+        root: ChannelContext[models.Promotion], info: ResolveInfo, **kwargs
     ):
-        qs = root.node.categories.all()
-        return create_connection_slice(qs, info, kwargs, CategoryCountableConnection)
+        def _get_categories(predicates):
+            if category_ids := predicates.get("categoryPredicate"):
+                qs = Category.objects.filter(id__in=category_ids)
```

Comment:
Shouldn't be dataloader?

Reply:
Same question for `Collection`, `Product` and `Variant` 👇

Reply:
I need to have queryset here in order to pass it to `create_connection_slice` function.